pkg/cmd/pulumi/stack/stack_select.go

```diff
@@ -71,6 +71,10 @@ func newStackSelectCmd() *cobra.Command {
 				stack = args[0]
 			}
 
+			if stack == "" && !cmdutil.Interactive() {
+				return errors.New("no stack name given, and cannot prompt in non-interactive mode. Use --stack to specify a stack name")
```

Comment:
```suggestion
				return errors.New("no stack name given, use --stack to specify a stack name")
```

I feel like the wording about not being able to prompt is just gonna be noise to most. Keep it short just tell them the fix.